### DISCUSSION TITLE
Safari / Mac does not play vp9 files

### DISCUSSION THREAD

**Discussion by Str1atum** [immich-app/immich#9535] [Q&A]
### The bug

I changed my recoding settings to

VP9
crf 31
preset medium
aac audio
720p
max bitrate 0
two-pass off

Videos don't play in latest Safari and even when I download them, Quicktime will not open them either.
In Chrome everything works fine.
Any recommended settings or is this a bug?

### The OS that Immich Server is running on

Ubuntu 24.04

### Version of Immich Server

v.105.1

### Version of Immich Mobile App

v.105

### Platform with the issue

- [ ] Server
- [X] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
standard
```

### Your .env content

```Shell
standard
```

### Reproduction steps

```bash
1.
2.
3.
...
```

**Comment by mertalev** [comment]
If videos play on Chrome, then this is a Safari / Quicktime limitation. I believe there’s a setting in Safari you can enable to make VP9 work. Not sure about Quicktime.

Also Big Sur made some improvements here if you’re on an older version.

**Comment by Str1atum** [comment]
I'm on 14.5 so that shouldn't be an issue. There's no longer a setting for VP9 AFAIK, so maybe someone knows some working settings.

**Reply by mertalev** [reply]
Can you play other VP9 videos not from immich?

**Reply by Str1atum** [reply]
for example this site works fine:

https://base-n.de/webm/VP9%20Sample.html

**Reply by mertalev** [reply]
For HEVC, there's a tag we need to set in the output for Apple devices to accept it. I wonder if there's something like that for VP9.

**Reply by mertalev** [reply]
Also, you mentioned using Quick Sync for transcoding? Can you try transcoding a video with CPU, clearing the browser cache and trying again?

**Reply by NicholasFlamy** [reply]
That might be it, I've tested on Safari on MacOS Sonoma 14.2.1 with VP9 which was CPU transcoded (since my 8th gen Intel CPU doesn't support Quicksync VP9 encoding) on immich. Also, my VM doesn't have a GPU passed through so it must be software decoding.

**Reply by Str1atum** [reply]
I played around a bit and tried CPU transcoding / hardware acceleration disabled. Strangely, this creates H.265 videos, despite having selected VP9 (triple checked now)

**Reply by NicholasFlamy** [reply]
It sounds like it might be playing the originals.

**Comment by NicholasFlamy** [comment]
Found this online:
![](https://i.imgur.com/wETQGD5.png)

**Comment by Str1atum** [comment]
that must be on some old version - it's integrated now and considered "stable"

![Screenshot 16 05 2024 um 18 41 28 PM](https://github.com/immich-app/immich/assets/76776216/a39d2427-a7ab-4d8f-8f33-274cc2b2d5df)

**Reply by NicholasFlamy** [reply]
I don't have a Mac but I have a MacOS Sonoma VM so maybe I can try to replicate the problem. I'll copy your settings to an immich instance of mine.

Also, the screenshot I sent I found online.

**Reply by NicholasFlamy** [reply]
Is the immich server using hardware acceleration for transcoding? Just trying to replicate your setup.

**Comment by Str1atum** [comment]
Yes, hardware transcoding is enabled

**Reply by NicholasFlamy** [reply]
Intel? Nvidia? AMD? (Not sure if that's even supported)

**Reply by Str1atum** [reply]
quicksync on Intel Iris Xe
